WcaOnRails/spec/features/sca_payments_spec.rb

```diff
@@ -37,18 +33,30 @@
 def fill_confirm_and_expect(card, button_id, message)
   fill_in with: "John Doe", id: "cardholder-name"
   within_frame(page.first("#card-element iframe")) do
-    fill_in with: card[:number], name: "cardnumber"
+    # WARNING, RANT! stripe.js has an input formatting mechanism that conflicts with JS test drivers.
+    # Pasting the entire CC number at once is completely non-deterministic and results in random gibberish every time.
+    # Interestingly, it *deterministically* screws up, i.e. the number gets entered incorrectly the same way
+    # when sending the individual characters of the credit card number one-by-one. As a potential workaround,
+    # I tried sending "right arrow" keys to the browser to skip to the end of the input with no avail.
```

Comment:
This is annoying but I can't say I'm surprised :(
I didn't try to find an alternative way but I'd assume it's fine as:
  - this weird behavior only happens in test
  - if this ever gets fixed it will be easy to see the test failing and rollback to the previous "regular" way of inputting the credit card number

Reply:
Agreed. I am confident that this is a specifc hiccup of the the test engine JS driver. Setting the engine to `headless: false` and manually punching in numbers into the Chrome window works as expected :relieved: